### PR TITLE
Promote image : devconsole operator

### PR DESCRIPTION
**Pull request by sbose78**

### CODE REVIEW

**Comment by sbose78** [comment]
@petr-muller @droslean Does this look good? This is a step towards publishing image on quay. Will do mirroring in a separate PR.

File: ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml [hunk 1]
```diff
@@ -25,6 +25,10 @@ images:
   dockerfile_path: openshift-ci/Dockerfile.deploy
   to: devconsole-operator
 
+promotion:
+  cluster: https://api.ci.openshift.org
+  namespace: ci
```
Comment (petr-muller) [inline ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: I don't think we want unrelated (I mean unrelated to the actual CI tooling) images to be promoted into the `ci` namespace... I think this should do what `knative` is doing and promote into an imagestream in `openshift`, or maybe have a NS of it's own, not sure.

@stevekuznetsov @smarterclayton any thoughts?

Comment (sbose78) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: Thanks for the feedback @petr-muller !

Yeah, knative does this:

```
cluster: https://api.ci.openshift.org
  namespace: openshift
  name: knative-v0.3
```

I could go with something like

```
cluster: https://api.ci.openshift.org
  namespace: openshift
  name: devconsole-0.1.0
```

Comment (sbose78) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: Done 
```
promotion:
  cluster: https://api.ci.openshift.org
  namespace: devconsole
  tag: latest
```

Comment (sbose78) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: https://github.com/openshift/release/pull/3357/commits/1429a01a5e54ab8ef04dd9efd0c2340a43958727

Comment (petr-muller) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: Looks good. You'll probably need to add some minimal config (to set up that namespace) to https://github.com/openshift/release/tree/master/projects

Comment (sbose78) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: Thanks @petr-muller , will do.

Comment (baijum) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: The README says the `projects` directory is deprecated.
https://github.com/openshift/release#component-project-build-manifests

Comment (stevekuznetsov) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: Is there a good reason to need a new namespace? Just use the first of your two suggestions and you need no extra config

Comment (sbose78) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: @stevekuznetsov You mean, this should do? 
```
cluster: https://api.ci.openshift.org
  namespace: openshift
  name: devconsole-0.1.0
```

**Comment by baijum** [comment]
@sbose78 Can you squash all commits into one commit?

**Comment by sbose78** [comment]
> @sbose78 Can you squash all commits into one commit?

@baijum , done!

**Comment by sbose78** [comment]
@droslean @petr-muller Could you please approve?

**Comment by droslean** [comment]
the imagestream `devconsole-0.1.0` doesn't exists. You should probably want to add the config to create it in [projects](https://github.com/openshift/release/tree/master/projects)

**Comment by sbose78** [comment]
@droslean , Sorry I am confused. The Readme says that `projects` are being deprecated. And then Steve pointed out that we might mot need any more config. https://github.com/openshift/release/pull/3357#discussion_r276382005


Could you please tell me what/where I should be adding config to?

**Comment by sbose78** [comment]
@pmorie Could you please take a look at this?

**Comment by baijum** [comment]
@sbose78 Please rebase and squash commits into a single commit.

**Pull request merged**